src/sage/features/fricas.py

```diff
@@ -0,0 +1,65 @@
+r"""
+Features for testing the presence of ``fricas``
+"""
+
+# *****************************************************************************
+#       Copyright (C) 2023 Dima Pasechnik
+#
+#  Distributed under the terms of the GNU General Public License (GPL)
+#  as published by the Free Software Foundation; either version 2 of
+#  the License, or (at your option) any later version.
+#                  https://www.gnu.org/licenses/
+# *****************************************************************************
+
+import os
+import subprocess
+from . import Executable, FeatureTestResult
+
+class FriCAS(Executable):
+    r"""
+    A :class:`~sage.features.Feature` which checks for the :ref:`fricas <fricas>` binary.
+
+    EXAMPLES::
+
+        sage: from sage.features.fricas import FriCAS
+        sage: FriCAS().is_present()  # optional - fricas
+        FeatureTestResult('fricas', True)
+    """
+    def __init__(self):
+        r"""
+        TESTS::
+
+            sage: from sage.features.fricas import FriCAS
+            sage: isinstance(FriCAS(), FriCAS)
+            True
+        """
+        Executable.__init__(self, name="fricas", spkg="fricas",
+                            executable="fricas",
+                            url="https://fricas.github.io")
+
+    def is_functional(self):
+        r"""
+        Check whether ``fricas`` works on trivial input.
+
+        EXAMPLES::
+
+            sage: from sage.features.fricas import FriCAS
+            sage: FriCAS().is_functional()  # optional - fricas
+            FeatureTestResult('fricas', True)
+        """
+        command = ["fricas -nosman", "2+2"]
+        try:
+            lines = subprocess.check_output(command, stderr=subprocess.STDOUT)
```

Comment:
if you don't use `shell=True`, this cannot work

Reply:
I still don't understand what we are doing here. FriCAS feature is something configured; the testing framework should **test the configuration**, and do  not engage in these run-time double-checks.

As soon as testing framework knows that FriCAS should be available, it must just ignore `# optional - fricas` tags,
and treat FriCAS as available.

I don't get why distributions have any need for these, either.